db/comdb2.c

```diff
@@ -1356,9 +1355,9 @@ static int clear_csc2_files(void)
             if (!strcmp(dp->d_name, ".") || !strcmp(dp->d_name, ".."))
                 continue;
 
-            ptr = strstr(dp->d_name, ".csc2");
+            ptr = strrchr(dp->d_name, '.');
 
-            if (ptr) {
+            if (ptr && (strncmp(ptr, ".csc2", sizeof(".csc2")) == 0)) {
```

Comment:
sizeof(".csc2") is 6 (includes '\0') so strncmp will compare 6 characters, so will make sure that ptr[5] == '\n', cool.